stix_shifter/stix_transmission/src/modules/bigfix/bigfix_status_connector.py

```diff
@@ -64,26 +67,27 @@ def create_status_connection(self, search_id):
                         if total_results <= 0:
                             return_obj['status'] = Status.ERROR.value
 
-                    if return_obj['success'] == True and return_obj['status'] == Status.COMPLETED.value:
+                    if return_obj['success'] and return_obj['status'] == Status.COMPLETED.value:
```

Comment:
do you have any query that sets the status COMPLETE? I have tried many queries all of them stayed RUNNING forever even though we have got results back already. UDS will automatically cancel the query when status stuck at RUNNING state. We need a bail out condition here to set the status to COMPLETE if we have received all the results that are available.  we can discuss further details if you need.

Reply:
Incorporated the changes that Danny had suggested in his mail.
Now internally the status query will check the status for a period of  (timeout - 6 seconds) (if timeout = 1min, will check for 54 sec) sec and if progress doesn't reach 100% will return back a status of **'Completed'** and along with the last observed progress percentage.

Reply:
As discussed and shared over email we have incorporated the query status completion logic with the latest commit.